### ISSUE TITLE
Not showing all test classes

### ISSUE THREAD

**Issue by ennorehling** [recca0120/vscode-phpunit#164]
The Test Explorer fails to pick up some tests when they extend other tests.

For example, I have the following test classes:
```
abstract class FooInterfaceTest extends TestCase {
    public function test_foo() {
        $this->assertTrue(true);
    }
}

final class FooTest extends FooInterfaceTest {
    public function test_bar() {
        $this->assertTrue(true);
    }
}

final class FakeFooTest extends FooInterfaceTest {
}
```
What I expect to see is both FakeFooTest FooTest with one and two tests respectively listed in the explorer. What I see instead is only the FooTest, and only the test_bar function that it implements, not the test(s) that it receives from the class that it extends.

NB: What I'm trying to do here is to test both a class and its testing double, without duplicating the tests.

**Comment by jlecordier** [comment]
Out of curiosity, does this work when using regular PHPUnit ?

**Comment by TimoLehnertz** [comment]
Yes it does. I have the same issue. I love the extension but half my tests wont get executed because of this

**Comment by recca0120** [comment]
@ennorehling does FakeFooTest have the setup method? if yes. I can make FakeFooTest list in test explorer.

**Comment by ennorehling** [comment]
I no longer have this problem, I rewrote my tests. But no, there was no setUp function.

**Comment by afcamps** [comment]
Hi @recca0120 ,

I had a similar issue. I forked the project to understand why some tests weren't showing up, and in my case, the engine `php-parser` was throwing errors due to new language features.

After setting the PHP version to `8.4`, all tests were displayed correctly.

I'm not sure if it's related to this current issue,
but as a possible workaround, it might help to make the PHP version configurable and pass it directly to the parser.

Best regards,
Aurélien

**Comment by recca0120** [comment]
Hi @afcamps,

Thanks for the insights! Could you make a PR for this issue when you have time? Also, it would be helpful if you could share some examples of the errors thrown by php-parser, so I can better understand the problem.

Thanks again!
